File: CodingSamples/Java/Language-Runtime/SubclassTest/demo1/payroll/Employee.java
package payroll;

public class Employee {
    
    private int hours;
    private float rate;

    public Employee(int h, float r) {
        hours = h;
        rate = r;
    }

    //paremeterless constructor
    public Employee() {
        this(180, 50); //calling another constructor of this class
    }

    //exposing hours field as a property using 
    //accessor (get/set) methods
    public int getHours() {
        return hours;
    }

    public void setHours(int value) {
        hours = value;
    }

    public float getRate() {
        return rate;
    }

    public void setRate(float value) {
        rate = value;
    }

    public double income() {
        double amount = hours * rate;
        int ot = hours - 180;
        if(ot > 0)
            amount += 50 * ot;
        return amount;
    }
}

File: CodingSamples/DotNet/Language-Runtime/ClassTypeTest/DemoApp1/Payroll/Employee.cs
namespace Payroll
{
    class Employee
    {
        private int hours;
        private float rate;

        public Employee(int h, float r)
        {
            hours = h;
            rate = r;
        }

        //parameterless contructor
        public Employee() : this(0, 0) {}

        //a property provides accessor (get/set) methods for a field
        //so that they can be used in an expression with a syntax
        //similar to direct accessing that field 
        public int Hours
        {
            get
            {
                return hours;
            }

            set
            {
                hours = value;
            }
        }

        public float HourlyRate
        {
            get { return rate; }
            set { rate = value; }
        }

        //a method declared with 'virtual' keyword can be
        //overridden in the derived class
        public virtual double Income()
        {
            double amount = hours * rate;
            int ot = hours - 180;
            if(ot > 0)
                amount += 50 * ot;
            return amount;
        }
    }
}